### ISSUE TITLE
'NoneType' object is not iterable

### ISSUE THREAD

**Issue by chocobuckle** [SublimeLinter/SublimeLinter#1072]
SublimeLinter no longer working and been getting the following error the last 24 hours...
```
Traceback (most recent call last):
File "lint.backend in C:\Users\Conor\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 97, in execute_lint_task
File "lint.linter in C:\Users\Conor\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 794, in lint
File "lint.linter in C:\Users\Conor\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 505, in get_cmd
TypeError: 'NoneType' object is not iterable
```
...tried reinstalling SublimeLinter and restarting ST3, but no joy. Any advise please? :)

**Comment by braver** [comment]
Please enable debug and print the full contents of the console after restarting ST and linting a file.

**Comment by chocobuckle** [comment]
Sorry, should have done that in the first place! Hmmm, it seems SublimeLinter-flow is actually the cause. If I don't have 'flow-bin' installed as a dependency I get the error, and as soon as I install 'flow-bin' the error goes away. Which is fine if you're planning to use Flow on a  project, but problematic if you're not...

```
DPI scale: 1
startup, version: 3143 windows x64 channel: stable
executable: /C/Program Files/Sublime Text 3/sublime_text.exe
working dir: /C/Program Files/Sublime Text 3
packages path: /C/Users/Conor/AppData/Roaming/Sublime Text 3/Packages
state path: /C/Users/Conor/AppData/Roaming/Sublime Text 3/Local
zip path: /C/Program Files/Sublime Text 3/Packages
zip path: /C/Users/Conor/AppData/Roaming/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.127431
startup time: 0.170431
first paint time: 0.174431
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.55-jsonschema
reloading plugin Character Table.Character Table
reloading plugin Colorsublime.colorsublime-plugin
reloading plugin Emmet.emmet-plugin
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin PackageResourceViewer.package_resource_viewer
reloading plugin PackageResourceViewer.package_resources
reloading plugin SublimeLinter-eslint.linter
reloading plugin SublimeLinter-flow.linter
reloading plugin SublimeLinter-json.linter
reloading plugin SublimeLinter.__init__
reloading plugin SublimeLinter.busy_indicator_view
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.goto_commands
reloading plugin SublimeLinter.highlight_view
reloading plugin SublimeLinter.log_handler
reloading plugin SublimeLinter.message_view
reloading plugin SublimeLinter.panel_view
reloading plugin SublimeLinter.status_bar_view
reloading plugin SublimeLinter.sublime_linter
reloading plugin SublimeLinter.tooltips_view
reloading plugin TrailingSpaces.trailing_spaces
reloading plugin Babel.Babel
plugins loaded
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 210, in on_api_ready
    m.plugin_loaded()
  File "Character Table in C:\Users\Conor\AppData\Roaming\Sublime Text 3\Installed Packages\Character Table.sublime-package", line 193, in plugin_loaded
  File "Character Table in C:\Users\Conor\AppData\Roaming\Sublime Text 3\Installed Packages\Character Table.sublime-package", line 188, in create_mnemonic_keymap
FileExistsError: [WinError 183] Cannot create a file when that file already exists: 'C:\\Users\\Conor\\AppData\\Roaming\\Sublime Text 3\\Packages\\User\\Character Table\\Default\\Default.sublime-keymap'
SublimeLinter: SublimeLinter.sublime-package:71: Logging installed; log level INFO
SublimeLinter: SublimeLinter.sublime-package:62: debug mode: on
SublimeLinter: SublimeLinter.sublime-package:63: version: 4.0.2
Emmet: No need to update PyV8
SublimeLinter: SublimeLinter.sublime-package:252: detected syntax: javascript
SublimeLinter: #1 SublimeLinter.sublime-package:782: 'eslint' is linting 'MainContainer.js'
SublimeLinter: #2 SublimeLinter.sublime-package:782: 'flow' is linting 'MainContainer.js'
SublimeLinter: #1 SublimeLinter.sublime-package:1178: eslint: MainContainer.js ['C:\\Users\\Conor\\atlas-of-remote-islands-react\\node_modules\\.bin\\eslint.cmd', '--format', 'compact', '--stdin', '--stdin-filename', 'C:\\Users\\Conor\\atlas-of-remote-islands-react\\src\\containers\\MainContainer.js']
SublimeLinter: #1 SublimeLinter.sublime-package:1181: eslint: cwd: C:\Users\Conor\atlas-of-remote-islands-react
SublimeLinter: #2 SublimeLinter.sublime-package:103: PATH:
    C:\Program Files (x86)\Common Files\Intel\Shared Libraries\redist\intel64_win\compiler
    C:\Program Files (x86)\Common Files\Intel\Shared Libraries\redist\ia32_win\compiler
    C:\WINDOWS\system32
    C:\WINDOWS
    C:\WINDOWS\System32\Wbem
    C:\WINDOWS\System32\WindowsPowerShell\v1.0\
    C:\Program Files\Git\cmd
    C:\Users\Conor\AppData\Roaming\npm
    C:\Program Files (x86)\Common Files\Acronis\VirtualFile\
    C:\Program Files (x86)\Common Files\Acronis\VirtualFile64\
    C:\Program Files (x86)\Common Files\Acronis\SnapAPI\
    C:\Program Files\nodejs\
    C:\Program Files (x86)\Yarn\bin
    C:\Users\Conor\AppData\Local\Microsoft\WindowsApps
    C:\Users\Conor\AppData\Roaming\npm
    C:\Users\Conor\AppData\Local\Yarn\bin
    C:\Program Files\Microsoft VS Code\bin

SublimeLinter: WARNING: flow cannot locate 'flow' 
SublimeLinter: flow attempting to run from: None 
SublimeLinter: #2 __init__.py:1273: ERROR: Linter crashed.

Traceback (most recent call last):
  File "lint.backend in C:\Users\Conor\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 97, in execute_lint_task
  File "lint.linter in C:\Users\Conor\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 794, in lint
  File "lint.linter in C:\Users\Conor\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 505, in get_cmd
TypeError: 'NoneType' object is not iterable
Package Control: Skipping automatic upgrade, last run at 2018-03-03 19:13:32, next run at 2018-03-03 20:13:32 or after
SublimeLinter: #1 SublimeLinter.sublime-package:809: eslint output:
    C:\Users\Conor\atlas-of-remote-islands-react\src\containers\MainContainer.js: line 122, col 1, Error - Line 122 exceeds the maximum line length of 100. (max-len)
    C:\Users\Conor\atlas-of-remote-islands-react\src\containers\MainContainer.js: line 235, col 9, Warning - Unexpected console statement. (no-console)
    C:\Users\Conor\atlas-of-remote-islands-react\src\containers\MainContainer.js: line 402, col 1, Error - Line 402 exceeds the maximum line length of 100. (max-len)

    3 problems
SublimeLinter: #1 SublimeLinter.sublime-package:850: eslint: No match for line: ''
SublimeLinter: #1 SublimeLinter.sublime-package:850: eslint: No match for line: '3 problems'

```

**Comment by kaste** [comment]
This is known issue with the flow linter. I made a PR here yesterday. Thanks for reporting anyway!

**Comment by chocobuckle** [comment]
Fixed in 4.5.0 release of SublimeLinter-flow, thanks @kaste :)
